lib/astutils.cpp

```diff
@@ -2119,11 +2119,14 @@ bool isVariableChanged(const Token *tok, int indirect, const Settings *settings,
     int derefs = 0;
     while (Token::simpleMatch(tok2->astParent(), "*") ||
            (Token::simpleMatch(tok2->astParent(), ".") && !Token::simpleMatch(tok2->astParent()->astParent(), "(")) ||
+           (Token::simpleMatch(tok2->astParent(), "&") && !tok2->astParent()->astOperand2() && Token::simpleMatch(tok2->astParent()->astParent(), ".") && tok2->astParent()->astParent()->originalName()=="->") ||
```

Comment:
I would somewhat prefer `tok2->astParent() && tok2->astParent()->isUnaryOp("&")` .. to hide the logic a bit how to check if it's a address-of unary operator.